storm/src/main/storm/mesos/MesosSupervisor.java

```diff
@@ -167,8 +189,26 @@ public void launchTask(ExecutorDriver driver, TaskInfo task) {
       driver.sendStatusUpdate(status);
     }
 
+    /**
+     * If a failure occurs we halt this process, to avoid having an inconsistency
+     * between Mesos's view of the running tasks and which processes are actually
+     * running.
+     * Killing this supervisor process also kills any child worker processes.
+     */
     @Override
     public void killTask(ExecutorDriver driver, TaskID id) {
+      int port = 0;
+      try {
+        port = MesosCommon.portFromTaskId(id.getValue());
+      } catch (IllegalArgumentException e) {
+        LOG.error("killTask: Halting executor process because we had a problem" +
+            " extracting the port from the TaskID: " + id.getValue(), e);
+        Runtime.getRuntime().halt(1);
+      }
+
+      LOG.info("killTask: killing task " + id.getValue() +
+          " which is running on port " + port);
+      _state.remove(Integer.toString(port));
```

Comment:
Unfortunately, this breaks `killedWorker()`.  The intent is to allow MesosSupervisor to communicate to storm-core supervisor that it no longer has this port assigned, and thus should kill the associated worker process.  Unfortunately, by removing the port from `_state`, we prevent `killedWorker()` from issuing the status update later (which needs the TaskID).  So I'm going to change from using `LocalState _state` for this [1] to just using a memory-based map pointing to a tuple <TaskID, Boolean> (boolean indicating whether the port is currently assigned here).  Then I'll change the methods that interact with `_state` to deal with that tuple.

[1] LocalState stores persistently into disk, which doesn't have any value for the MesosSupervisor, since it cannot be restarted.